--- web-base/service-worker.js	2026-05-24 09:23:17
+++ web-base/service-worker-patched.js	2026-05-24 09:31:47
@@ -4,10 +4,10 @@
 var LANGUAGE = 'en-GB';
 var CACHE_NAME = 'Lands-V'+VERSION;
 var FILES_TO_CACHE = [
-	'/',
+	fan_base_url_prefix+'/',
 
 	// CSS
-	'/css/style.css?v='+VERSION,
+	fan_base_url_prefix+'/css/style.css?v='+VERSION,
 ];
 
 self.addEventListener('install', function(event) {
@@ -21,15 +21,11 @@
 	var ENCYCLOPEDIA = swURL.searchParams.get('encyclopedia') ?? 0;
 
 	// Determine asset url (php script that takes language into account)
-	var assetURL = '/assets.php';
-	assetURL += '?language='+LANGUAGE;
-	assetURL += '&audio-format='+AUDIOFORMAT;
-	assetURL += '&encyclopedia='+ENCYCLOPEDIA;
-	assetURL += '&version='+VERSION;
+	var assetURL = AUDIOFORMAT == '.' ? fan_base_url_prefix+'/assets.json' : fan_base_url_prefix+'/assets-'+AUDIOFORMAT+'.json';
 
 	// Add possible language version index.html to cache, like /es
 	if (LANGUAGE != 'en-GB') {
-		FILES_TO_CACHE.push('/' + LANGUAGE.split('-')[0]);
+		FILES_TO_CACHE.push(fan_base_url_prefix + '/' + LANGUAGE.split('-')[0]);
 	}
 
 	// Add a possible dev url with parameters to the files to cache, like /?devmode=0
@@ -37,10 +33,10 @@
 	var searchParams = swURL.searchParams.get('url-parameters');
 	if (searchParams !== '' && searchParams != null) {
 		var URLParameters = decodeURIComponent(searchParams);
-		FILES_TO_CACHE.push('/' + URLParameters);
+		FILES_TO_CACHE.push(fan_base_url_prefix + '/' + URLParameters);
 		if (LANGUAGE != 'en-GB') {
 			// The same for other language versions
-			FILES_TO_CACHE.push('/' + LANGUAGE.split('-')[0] + URLParameters);
+			FILES_TO_CACHE.push(fan_base_url_prefix + '/' + LANGUAGE.split('-')[0] + URLParameters);
 		}
 	}
 
